coreneuron/network/multisend_setup.cpp

```diff
@@ -421,8 +421,10 @@ static void fill_multisend_lists(bool use_phase2,
         PreSyn* ps = g.second;
         if (ps->output_index_ >= 0) {  // only ones that generate spikes
             int i = ps->multisend_index_;
-            max_ntarget_host = std::max(targets_phase1[i], max_ntarget_host);
-            max_multisend_targets = std::max(targets_phase1[i + 1], max_multisend_targets);
+            if (i >= 0) {
```

Comment:
OK! It seems that in the failing test, the input vector `r` is empty but the `gid2out` map is not.

Reply:
I convinced myself that this is a correct change by running the tqperf test with some extra cells that were not sources for cells on other ranks. I.e. cell spikes do not get transferred  to other ranks if there are no targets for them on the other ranks.
I think it would be a good idea add the comment
```if (i >= 0) { // only if the gid has targets on other ranks.```